src/hotspot/share/nmt/virtualMemoryTracker.cpp

```diff
@@ -361,7 +361,7 @@ bool VirtualMemoryTracker::add_reserved_region(address base_addr, size_t size,
       // It can happen when the regions are thread stacks, as JNI
       // thread does not detach from VM before exits, and leads to
       // leak JavaThread object
-      if (reserved_rgn->mem_tag() == mtThreadStack) {
+      if ((reserved_rgn->mem_tag() == mtThreadStack) && (reserved_rgn->mem_tag() == mem_tag)) {
```

Comment:
Please explain the changes in this function, especially the changed conditions. The added conditions for tag equality seem wrong to me.

Reply:
Sorry, I think this was just a debugging code that I left it by mistake - I was testing whether we can tighten up the logic of when we allow the tags to be allowed to change, but this looks indeed wrong, since nothing would change.

Reverted.